manager/controllers/app/blueprint_controller.go

```diff
@@ -141,6 +155,42 @@ func (r *BlueprintReconciler) applyChartResource(log logr.Logger, chartSpec app.
 	nbytes, _ := yaml.Marshal(args)
 	log.Info(fmt.Sprintf("--- Values.yaml ---\n\n%s\n\n", nbytes))
 
+	if chartSpec.ChartPullSecret != "" {
+		// obtain ChartPullSecret
+		pullSecret := corev1.Secret{}
+		pullSecrets := []corev1.Secret{}
+
+		if err := r.Get(ctx, types.NamespacedName{Namespace: utils.GetSystemNamespace(), Name: chartSpec.ChartPullSecret},
+			&pullSecret); err == nil {
+			// if this is not a dockerconfigjson, ignore
+			if pullSecret.Type == "kubernetes.io/dockerconfigjson" {
+				pullSecrets = append(pullSecrets, pullSecret)
+			}
+		}
+
+		// create a keyring of all dockerconfigjson secrets, to be used for lookup
+		keyring := credentialprovider.NewDockerKeyring()
+		keyring, _ = credentialprovidersecrets.MakeDockerKeyring(pullSecrets, keyring)
+		repoToPull, err := getDomainFromImageName(chartSpec.Name)
+		if err != nil {
+			return ctrl.Result{}, errors.WithMessage(err, chartSpec.Name+": failed to parse image name")
+		}
+
+		creds, withCredentials := keyring.Lookup(chartSpec.Name)
+		if withCredentials {
+			for _, cred := range creds {
+				err := r.Helmer.RegistryLogin(repoToPull, cred.Username, cred.Password, false)
```

Comment:
Does RegistryLogout is also needed at the end?

Reply:
good point. I will address this issue.